### DISCUSSION TITLE
Dockerfile manager doesn't understand `rc` versioning

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#20107] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.97.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

When using a `Dockerfile`:

```dockerfile
FROM whitesource/renovate:3.0.0-rc.0 as renovate
# ...
```

And using Renovate, there was no prompting to upgrade from `3.0.0-rc.0` to `3.0.0` when it was [released last week](https://github.com/mend/renovate-on-prem/issues/248#issuecomment-1406566167).

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: packageFiles with updates (repository=..., baseBranch=main)

               {
                 "depName": "whitesource/renovate",
                 "currentValue": "3.0.0-rc.0",
                 "replaceString": "whitesource/renovate:3.0.0-rc.0",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "depIndex": 1,
                 "updates": [],
                 "warnings": [],
                 "versioning": "docker",
                 "sourceUrl": "https://github.com/whitesource/renovate-on-prem",
                 "registryUrl": "https://index.docker.io",
                 "changelogUrl": "https://github.com/whitesource/renovate-on-prem",
                 "currentVersion": "3.0.0",
                 "fixedVersion": "3.0.0-rc.0"
               }
```

</details>

### Have you created a minimal reproduction repository?

https://github.com/jamietanna/renovate-iss-20106

**Comment by rarkins** [comment, marked as answer]
This is not a bug, so converted to a discussion.

Managers don't understand versioning at all, that's up to `versioning` modules. The default `docker` versioning for `docker` datasource treats suffixes as "compatibility", e.g. `-alpine`. The majority of Docker images are _not_ semver compliant because of the use of such suffixes in tags.

If the Docker image you use supports semver then you can configure `versioning=semver` in a package rule.

Potentially in future there could be a _feature_ to try to "automatically" detect semver-compliant image tags and use semver instead of docker.

**Reply by jamietanna** [reply]
Oh awesome, thanks Rhys for the explanation 🙌 

Looks like this is the rule I needed to add for this specific case:

```json
{
  "packageRules": [
    {
      "matchPackageNames": ["whitesource/renovate"],
      "versioning": "semver"
    }
  ]
}
```
